Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -372,6 +374,17 @@ void FullscreenManager::exitFullscreen(RefPtr<DeferredPromise>&& promise)
             return;
         }
 
+        // Run the "fully unlock the screen orientation" steps with doc.
+        // https://w3c.github.io/screen-orientation/#dfn-fully-unlock-the-screen-orientation-steps
+        if (auto orientationManager = page->screenOrientationManager()) {
+            if (auto orientationPromise = orientationManager->takeLockPromise()) {
+                m_document.eventLoop().queueTask(TaskSource::UserInteraction, [orientationPromise = WTFMove(orientationPromise)]() {
+                    if (orientationPromise)
+                        orientationPromise->reject(Exception { AbortError, "Exiting fullscreen aborted orientation change"_s });
+                });
+            }
+        }
+
```

Comment:
This should move as the first thing in `finishExitFullscreen()` as this is where we actually change the fullscreenElement.

Also, you'll want to wrap this with:
```
if (document->settings().fullscreenRequirementForScreenOrientationLockingEnabled()) {
```